src/core/ext/filters/client_channel/lb_policy/xds/xds_routing.cc

```diff
@@ -227,6 +228,10 @@ XdsRoutingLb::PickResult XdsRoutingLb::RoutePicker::Pick(PickArgs args) {
   }
   std::vector<absl::string_view> path_elements =
       absl::StrSplit(path.substr(1), '/');
+  static LazyRE2 hostportish_re = {"([^:{}\\s,]+):([0-9]+)"};
+  if (RE2::FullMatch(path.data(), *hostportish_re)) {
+    gpr_log(GPR_INFO, "test re2");
```

Comment:
this looks like a test-only log, are you really planning to bring this in or is this just a leftover from your experiments?

Reply:
+1.  We don't really have any use for RE2 in this code without the code from #22951.

The easiest way of handling this might be to remove the regex support from #22951 and get that PR merged first.  Then this PR can add the RE2 dependency and the regex matcher support at the same time.